src/plugin.tsx

```diff
@@ -20,7 +20,9 @@ export const mediaAssetSource = {
 
 const tool = {
   ...plugin,
-  component: Tool
+  component: Tool,
+  // @ts-expect-error TS doesn't know about this internal field see https://github.com/sanity-io/sanity/pull/7980
+  __internalApplicationType: 'sanity/media'
 } satisfies SanityTool
```

Comment:
An alternative approach to avoid `@ts-expect-error` would be to extend the satisfies clause to the following:
```
satisfies SanityTool & {__internalApplicationType: 'sanity/media'}
```

That way, the `__internalApplicationType` property with its value will be expected.